### PR TITLE
🚀 [Story performance] Add css to show story from beginning

### LINKED ISSUE #36050: [Story performance] Show story as soon as the JS loads, without waiting for the runtime build callbacks
Currently the components are hidden until the story builds (and layouts). We want to show the story ASAP so we'll need to override some of the AMP CSS rules that hide the elements that are not created.

### PR DESCRIPTION
**Pull request by mszylkowski**
Closes #36050

Adds styles to display the first page of the story before the JS loads.

### CODE REVIEW

File: extensions/amp-story/1.0/amp-story.css [hunk 1]
```diff
@@ -244,6 +252,10 @@ amp-story-page[distance="2"] {
   transform: translateY(100vh) !important;
 }
 
+amp-story:not(.i-amphtml-story-loaded) amp-story-page:not([active]) {
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/amp-story.css:255]: Plz add a short comment

File: extensions/amp-story/1.0/amp-story.css [hunk 2]
```diff
@@ -279,6 +291,13 @@ amp-story-grid-layer {
   pointer-events: none !important;
 }
 
+:root:not(.i-amphtml-story-vertical) amp-story, amp-story:not(.i-amphtml-story-loaded) amp-story-page:first-of-type,
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/amp-story.css:294]: Plz add a less short comment :))

Comment (mszylkowski) [reply extensions/amp-story/1.0/amp-story.css]: I was able to simplify the CSS rules here so the comment doesn't have to be that long anymore 😁

File: extensions/amp-story/1.0/amp-story.css [hunk 3]
```diff
@@ -279,6 +291,13 @@ amp-story-grid-layer {
...
+amp-story:not(.i-amphtml-story-loaded) amp-story-page:first-of-type amp-story-grid-layer {
+  position: absolute !important;
+  top: 0;
+  left: 0;
+}
```
[the remark this answers is not in the document]
Comment (gmajoulet) [reply extensions/amp-story/1.0/amp-story.css:295]: We don't really have any guarantee that `ampshared` is loaded if we inline this code in the HEAD of the document. We should add these statements.

File: extensions/amp-story/1.0/amp-story.css [hunk 4]
```diff
@@ -224,7 +232,7 @@ amp-story-page:not(:first-of-type):not([distance]):not([active]) {
   transform: translateY(1000vh) !important;
 }
 
-amp-story-page:not([active]) {
+amp-story.i-amphtml-story-loaded amp-story-page:not([active]) {
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/amp-story.css:235]: Why do you need this change?

File: extensions/amp-story/1.0/amp-story.css (hunk 3 above)
[the remark this answers is not in the document]
Comment (mszylkowski) [reply extensions/amp-story/1.0/amp-story.css:295]: That's a good point, I missed that https://github.com/ampproject/amphtml/blob/main/extensions/amp-story/1.0/amp-story.css#L76 also provides them, so we're good. I don't think there are other styles we need from there, but it opens up the conversation of whether we'd need ampshared.css on stories, or we can save up those extra bytes

File: extensions/amp-story/1.0/amp-story.css (hunk 4 above)
Comment (gmajoulet) [reply extensions/amp-story/1.0/amp-story.css:235]: As seen offline, let's revisit this and only rely on `[distance]` and `[active]` attr

**Pull request merged**